### DISCUSSION TITLE
Phase properties into phase envelope

### DISCUSSION THREAD

**Discussion by caio-ramos** [CoolProp/CoolProp#2312] [General]
Hi everyone,

Can I calculate the vapor and liquid properties of a fluid in the phase envelope zone?
For example, I'm trying to calculate the gas viscosity and liquid viscosity at a (P,H) point in the phase envelope. I could be wrong, but so far I have been able to calculate the properties of the entire mixture, not each phase.

**Comment by ibell** [comment]
You should not consider mixture viscosity in CoolProp. The methods
implemented are not accurate.

**Reply by caio-ramos** [reply]
Viscosity was just an example, I actually want to calculate all the properties of each phase, rather than the properties of the mixture. Is there a way to do this?

**Comment by ibell** [comment]
Yes, you can use the saturated_liquid_keyed_output and
saturated_vapop_keyed_output functions from the low-level interface
